### PR TITLE
Extract createVectorHashers() helper function

### PR DESCRIPTION
**Pull request by mbasmanova**

### CODE REVIEW

File: velox/exec/HashAggregation.cpp [hunk 1]
```diff
@@ -51,17 +51,9 @@ HashAggregation::HashAggregation(
 
   auto inputType = aggregationNode->sources()[0]->outputType();
 
-  auto numHashers = aggregationNode->groupingKeys().size();
-  std::vector<std::unique_ptr<VectorHasher>> hashers;
-  hashers.reserve(numHashers);
-  for (const auto& key : aggregationNode->groupingKeys()) {
-    auto channel = exprToChannel(key.get(), inputType);
-    VELOX_CHECK_NE(
```
Comment (Yuhta) [inline velox/exec/HashAggregation.cpp:59]: We are losing this check

File: velox/exec/HashProbe.cpp [hunk 2]
```diff
@@ -128,13 +128,7 @@ HashProbe::HashProbe(
 
   auto numKeys = joinNode_->leftKeys().size();
   keyChannels_.reserve(numKeys);
-  hashers_.reserve(numKeys);
-  for (auto& key : joinNode_->leftKeys()) {
-    auto channel = exprToChannel(key.get(), probeType_);
-    keyChannels_.emplace_back(channel);
```
Comment (Yuhta) [inline velox/exec/HashProbe.cpp:134]: `keyChannels_` is not populated anymore

File: velox/exec/RowNumber.cpp [hunk 3]
```diff
@@ -34,19 +34,8 @@ RowNumber::RowNumber(
   const auto numKeys = keys.size();
 
   if (numKeys > 0) {
-    std::vector<std::unique_ptr<VectorHasher>> hashers;
-    hashers.reserve(numKeys);
-    for (const auto& key : keys) {
-      const auto channel = exprToChannel(key.get(), inputType);
-      VELOX_USER_CHECK_NE(
```
Comment (Yuhta) [inline velox/exec/RowNumber.cpp:41]: We lose this check

File: velox/exec/HashAggregation.cpp (hunk 1 above)
Comment (mbasmanova) [reply velox/exec/HashAggregation.cpp:59]: This check was redundant to begin with. exprToChannel never returns constant channel for a field reference.

File: velox/exec/HashProbe.cpp (hunk 2 above)
Comment (mbasmanova) [reply velox/exec/HashProbe.cpp:134]: That's a good point. Let me fix. Thanks.

**Review by Yuhta — APPROVED** [review]

**Pull request closed without merging**